Bug: cargo install crashes

### What happened?

I can't install reductstore by using cargo.

### Version

1.4.0-alpha.1

### OS

Linux

### Snippet to repeat the bug

```
cargo install --version 1.4.0-alpha.1 reductstore
```

### Logs

```Shell
error: environment variable `OUT_DIR` not defined at compile time
 --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/auth/proto.rs:6:18
  |
6 | include!(concat!(env!("OUT_DIR"), "/reduct.proto.auth.rs"));
  |                  ^^^^^^^^^^^^^^^
  |
  = help: Cargo sets build script variables at run time. Use `std::env::var("OUT_DIR")` instead
  = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)

error: environment variable `OUT_DIR` not defined at compile time
 --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/proto.rs:7:18
  |
7 | include!(concat!(env!("OUT_DIR"), "/reduct.proto.storage.rs"));
  |                  ^^^^^^^^^^^^^^^
  |
  = help: Cargo sets build script variables at run time. Use `std::env::var("OUT_DIR")` instead
  = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)

error: environment variable `COMMIT` not defined at compile time
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:53:9
   |
53 |         env!("COMMIT"),
   |         ^^^^^^^^^^^^^^
   |
   = help: Use `std::env::var("COMMIT")` to read the variable at run time
   = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)

error: environment variable `BUILD_TIME` not defined at compile time
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:54:9
   |
54 |         env!("BUILD_TIME")
   |         ^^^^^^^^^^^^^^^^^^
   |
   = help: Use `std::env::var("BUILD_TIME")` to read the variable at run time
   = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)

error: couldn't read /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/asset/console.zip: No such file or directory (os error 2)
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:77:39
   |
77 |         console: ZipAssetManager::new(include_bytes!("asset/console.zip")),
   |                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `include_bytes` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0432]: unresolved import `crate::auth::proto::Token`
 --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/auth/policy.rs:7:5
  |
7 | use crate::auth::proto::Token;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^ no `Token` in `auth::proto`

error[E0432]: unresolved import `crate::auth::proto::token`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/auth/token_repository.rs:16:25
   |
16 | use crate::auth::proto::token::Permissions;
   |                         ^^^^^ could not find `token` in `proto`

error[E0432]: unresolved imports `crate::auth::proto::Token`, `crate::auth::proto::TokenCreateResponse`, `crate::auth::proto::TokenRepo`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/auth/token_repository.rs:17:26
   |
17 | use crate::auth::proto::{Token, TokenCreateResponse, TokenRepo};
   |                          ^^^^^  ^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^ no `TokenRepo` in `auth::proto`
   |                          |      |
   |                          |      no `TokenCreateResponse` in `auth::proto`
   |                          no `Token` in `auth::proto`

error[E0432]: unresolved import `crate::storage::proto::bucket_settings`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/http_frontend/bucket_api.rs:21:28
   |
21 | use crate::storage::proto::bucket_settings::QuotaType;
   |                            ^^^^^^^^^^^^^^^ could not find `bucket_settings` in `proto`

error[E0432]: unresolved import `crate::storage::proto::BucketSettings`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/http_frontend/bucket_api.rs:22:5
   |
22 | use crate::storage::proto::BucketSettings;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `BucketSettings` in `storage::proto`

error[E0432]: unresolved import `crate::storage::proto::FullBucketInfo`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/http_frontend/bucket_api.rs:23:5
   |
23 | use crate::storage::proto::FullBucketInfo;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `FullBucketInfo` in `storage::proto`

error[E0432]: unresolved import `crate::storage::proto::QueryInfo`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/http_frontend/entry_api.rs:31:5
   |
31 | use crate::storage::proto::QueryInfo;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `QueryInfo` in `storage::proto`

error[E0432]: unresolved import `crate::auth::proto::Token`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/http_frontend/server_api.rs:16:5
   |
16 | use crate::auth::proto::Token;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^ no `Token` in `auth::proto`

error[E0432]: unresolved import `crate::storage::proto::bucket_settings`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/http_frontend/server_api.rs:20:28
   |
20 | use crate::storage::proto::bucket_settings::QuotaType;
   |                            ^^^^^^^^^^^^^^^ could not find `bucket_settings` in `proto`

error[E0432]: unresolved imports `crate::storage::proto::BucketInfoList`, `crate::storage::proto::ServerInfo`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/http_frontend/server_api.rs:21:29
   |
21 | use crate::storage::proto::{BucketInfoList, ServerInfo};
   |                             ^^^^^^^^^^^^^^  ^^^^^^^^^^ no `ServerInfo` in `storage::proto`
   |                             |
   |                             no `BucketInfoList` in `storage::proto`

error[E0432]: unresolved import `crate::auth::proto::token`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/http_frontend/token_api.rs:18:25
   |
18 | use crate::auth::proto::token::Permissions;
   |                         ^^^^^ could not find `token` in `proto`

error[E0432]: unresolved imports `crate::auth::proto::Token`, `crate::auth::proto::TokenCreateResponse`, `crate::auth::proto::TokenRepo`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/http_frontend/token_api.rs:19:26
   |
19 | use crate::auth::proto::{Token, TokenCreateResponse, TokenRepo};
   |                          ^^^^^  ^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^ no `TokenRepo` in `auth::proto`
   |                          |      |
   |                          |      no `TokenCreateResponse` in `auth::proto`
   |                          no `Token` in `auth::proto`

error[E0432]: unresolved import `crate::storage::proto::bucket_settings`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/bucket.rs:19:28
   |
19 | use crate::storage::proto::bucket_settings::QuotaType;
   |                            ^^^^^^^^^^^^^^^ could not find `bucket_settings` in `proto`

error[E0432]: unresolved imports `crate::storage::proto::BucketInfo`, `crate::storage::proto::BucketSettings`, `crate::storage::proto::EntryInfo`, `crate::storage::proto::FullBucketInfo`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/bucket.rs:20:29
   |
20 | ...roto::{BucketInfo, BucketSettings, EntryInfo, FullBucketInfo};
   |           ^^^^^^^^^^  ^^^^^^^^^^^^^^  ^^^^^^^^^  ^^^^^^^^^^^^^^ no `FullBucketInfo` in `storage::proto`
   |           |           |               |
   |           |           |               no `EntryInfo` in `storage::proto`
   |           |           no `BucketSettings` in `storage::proto`
   |           no `BucketInfo` in `storage::proto`

error[E0432]: unresolved imports `crate::storage::proto::record`, `crate::storage::proto::Block`, `crate::storage::proto::EntryInfo`, `crate::storage::proto::Record`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/entry.rs:10:29
   |
10 | ...roto::{record, ts_to_us, us_to_ts, Block, EntryInfo, Record};
   |           ^^^^^^                      ^^^^^  ^^^^^^^^^  ^^^^^^ no `Record` in `storage::proto`
   |           |                           |      |
   |           |                           |      no `EntryInfo` in `storage::proto`
   |           |                           no `Block` in `storage::proto`
   |           no `record` in `storage::proto`
   |
   = help: consider importing this struct instead:
           log::Record

error[E0432]: unresolved imports `crate::storage::proto::BucketInfoList`, `crate::storage::proto::BucketSettings`, `crate::storage::proto::Defaults`, `crate::storage::proto::ServerInfo`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/storage.rs:15:29
   |
15 | ...roto::{BucketInfoList, BucketSettings, Defaults, ServerInfo};
   |           ^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^  ^^^^^^^^  ^^^^^^^^^^ no `ServerInfo` in `storage::proto`
   |           |               |               |
   |           |               |               no `Defaults` in `storage::proto`
   |           |               no `BucketSettings` in `storage::proto`
   |           no `BucketInfoList` in `storage::proto`

error[E0432]: unresolved imports `crate::storage::proto::record`, `crate::storage::proto::Block`, `crate::storage::proto::Record`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/query/historical.rs:13:29
   |
13 | ...::{record::State as RecordState, ts_to_us, us_to_ts, Block, Record};
   |       ^^^^^^                                            ^^^^^  ^^^^^^ no `Record` in `storage::proto`
   |       |                                                 |
   |       could not find `record` in `proto`                no `Block` in `storage::proto`
   |
   = help: consider importing this struct instead:
           log::Record

error[E0432]: unresolved import `crate::storage::proto::Block`
 --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/reader.rs:7:39
  |
7 | use crate::storage::proto::{ts_to_us, Block};
  |                                       ^^^^^ no `Block` in `storage::proto`

error[E0432]: unresolved imports `crate::storage::proto::record`, `crate::storage::proto::Block`
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/writer.rs:14:29
   |
14 | use crate::storage::proto::{record, ts_to_us, Block};
   |                             ^^^^^^            ^^^^^ no `Block` in `storage::proto`
   |                             |
   |                             no `record` in `storage::proto`

error[E0412]: cannot find type `Block` in this scope
  --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/block_manager.rs:86:17
   |
86 |         block: &Block,
   |                 ^^^^^ not found in this scope

error[E0412]: cannot find type `Block` in this scope
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/block_manager.rs:120:17
    |
120 |         block: &Block,
    |                 ^^^^^ not found in this scope

error[E0412]: cannot find type `Block` in this scope
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/block_manager.rs:202:45
    |
202 |     fn load(&self, block_id: u64) -> Result<Block, HttpError>;
    |                                             ^^^^^ not found in this scope
    |
help: you might be missing a type parameter
    |
193 | pub trait ManageBlock<Block> {
    |                      +++++++

error[E0412]: cannot find type `Block` in this scope
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/block_manager.rs:213:28
    |
213 |     fn save(&self, block: &Block) -> Result<(), HttpError>;
    |                            ^^^^^ not found in this scope

error[E0412]: cannot find type `Block` in this scope
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/block_manager.rs:225:67
    |
225 | ...ze: u64) -> Result<Block, HttpError>;
    |                       ^^^^^ not found in this scope
    |
help: you might be missing a type parameter
    |
193 | pub trait ManageBlock<Block> {
    |                      +++++++

error[E0412]: cannot find type `Block` in this scope
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/block_manager.rs:236:30
    |
236 |     fn finish(&self, block: &Block) -> Result<(), HttpError>;
    |                              ^^^^^ not found in this scope

error[E0412]: cannot find type `Block` in this scope
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/block_manager.rs:243:45
    |
243 |     fn load(&self, block_id: u64) -> Result<Block, HttpError> {
    |                                             ^^^^^ not found in this scope
    |
help: you might be missing a type parameter
    |
242 | impl<Block> ManageBlock for BlockManager {
    |     +++++++

error[E0412]: cannot find type `Block` in this scope
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/block_manager.rs:253:28
    |
253 |     fn save(&self, block: &Block) -> Result<(), HttpError> {
    |                            ^^^^^ not found in this scope

error[E0412]: cannot find type `Block` in this scope
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/block_manager.rs:267:67
    |
267 | ...ze: u64) -> Result<Block, HttpError> {
    |                       ^^^^^ not found in this scope
    |
help: you might be missing a type parameter
    |
242 | impl<Block> ManageBlock for BlockManager {
    |     +++++++

error[E0412]: cannot find type `Block` in this scope
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/block_manager.rs:280:30
    |
280 |     fn finish(&self, block: &Block) -> Result<(), HttpError> {
    |                              ^^^^^ not found in this scope

error[E0433]: failed to resolve: use of undeclared type `Block`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/block_manager.rs:248:9
    |
248 |         Block::decode(Bytes::from(buf)).map_err(|e| {
    |         ^^^^^ use of undeclared type `Block`

error[E0433]: failed to resolve: use of undeclared type `Block`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/storage/block_manager.rs:268:25
    |
268 |         let mut block = Block::default();
    |                         ^^^^^ use of undeclared type `Block`

Some errors have detailed explanations: E0412, E0432, E0433.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `reductstore` due to 33 previous errors
warning: build failed, waiting for other jobs to finish...
error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {ServerApi::info}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:100:17
    |
100 |             get(ServerApi::info),
    |             --- ^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {ServerApi::info}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `axum::routing::get`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:403:1
    |
403 | top_level_handler_fn!(get, GET);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `get`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {ServerApi::list}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:104:17
    |
104 |             get(ServerApi::list),
    |             --- ^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {ServerApi::list}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `axum::routing::get`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:403:1
    |
403 | top_level_handler_fn!(get, GET);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `get`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {ServerApi::me}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:106:60
    |
106 |         .route(&format!("{}api/v1/me", api_base_path), get(ServerApi::me))
    |                                                        --- ^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {ServerApi::me}`
    |                                                        |
    |                                                        required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `axum::routing::get`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:403:1
    |
403 | top_level_handler_fn!(get, GET);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `get`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {TokenApi::token_list}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:114:17
    |
114 |             get(TokenApi::token_list),
    |             --- ^^^^^^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {TokenApi::token_list}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `axum::routing::get`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:403:1
    |
403 | top_level_handler_fn!(get, GET);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `get`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap, [type error]) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {TokenApi::create_token}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:118:18
    |
118 |             post(TokenApi::create_token),
    |             ---- ^^^^^^^^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap, [type error]) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {TokenApi::create_token}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `post`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:407:1
    |
407 | top_level_handler_fn!(post, POST);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `post`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {TokenApi::get_token}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:122:17
    |
122 |             get(TokenApi::get_token),
    |             --- ^^^^^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {TokenApi::get_token}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `axum::routing::get`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:403:1
    |
403 | top_level_handler_fn!(get, GET);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `get`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap) -> impl futures_util::Future<Output = Result<(), HttpError>> {TokenApi::remove_token}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:126:20
    |
126 |             delete(TokenApi::remove_token),
    |             ------ ^^^^^^^^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap) -> impl futures_util::Future<Output = Result<(), HttpError>> {TokenApi::remove_token}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `delete`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:402:1
    |
402 | top_level_handler_fn!(delete, DELETE);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `delete`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {BucketApi::get_bucket}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:131:17
    |
131 |             get(BucketApi::get_bucket),
    |             --- ^^^^^^^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {BucketApi::get_bucket}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `axum::routing::get`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:403:1
    |
403 | top_level_handler_fn!(get, GET);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `get`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap) -> impl futures_util::Future<Output = Result<(), HttpError>> {BucketApi::head_bucket}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:135:18
    |
135 |             head(BucketApi::head_bucket),
    |             ---- ^^^^^^^^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap) -> impl futures_util::Future<Output = Result<(), HttpError>> {BucketApi::head_bucket}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `head`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:404:1
    |
404 | top_level_handler_fn!(head, HEAD);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `head`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap, [type error]) -> impl futures_util::Future<Output = Result<(), HttpError>> {BucketApi::create_bucket}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:139:18
    |
139 |             post(BucketApi::create_bucket),
    |             ---- ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap, [type error]) -> impl futures_util::Future<Output = Result<(), HttpError>> {BucketApi::create_bucket}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `post`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:407:1
    |
407 | top_level_handler_fn!(post, POST);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `post`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap, [type error]) -> impl futures_util::Future<Output = Result<(), HttpError>> {BucketApi::update_bucket}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:143:17
    |
143 |             put(BucketApi::update_bucket),
    |             --- ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap, [type error]) -> impl futures_util::Future<Output = Result<(), HttpError>> {BucketApi::update_bucket}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `axum::routing::put`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:408:1
    |
408 | top_level_handler_fn!(put, PUT);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `put`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap) -> impl futures_util::Future<Output = Result<(), HttpError>> {BucketApi::remove_bucket}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:147:20
    |
147 |             delete(BucketApi::remove_bucket),
    |             ------ ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<std::string::String>, HeaderMap) -> impl futures_util::Future<Output = Result<(), HttpError>> {BucketApi::remove_bucket}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `delete`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:402:1
    |
402 | top_level_handler_fn!(delete, DELETE);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `delete`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, HeaderMap, axum::extract::Path<HashMap<std::string::String, std::string::String>>, axum::extract::Query<HashMap<std::string::String, std::string::String>>, BodyStream) -> impl futures_util::Future<Output = Result<(), HttpError>> {EntryApi::write_record}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:152:18
    |
152 |             post(EntryApi::write_record),
    |             ---- ^^^^^^^^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, HeaderMap, axum::extract::Path<HashMap<std::string::String, std::string::String>>, axum::extract::Query<HashMap<std::string::String, std::string::String>>, BodyStream) -> impl futures_util::Future<Output = Result<(), HttpError>> {EntryApi::write_record}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `post`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:407:1
    |
407 | top_level_handler_fn!(post, POST);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `post`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<HashMap<std::string::String, std::string::String>>, axum::extract::Query<HashMap<std::string::String, std::string::String>>, HeaderMap) -> impl futures_util::Future<Output = Result<impl IntoResponse, HttpError>> {EntryApi::read_record}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:156:17
    |
156 |             get(EntryApi::read_record),
    |             --- ^^^^^^^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<HashMap<std::string::String, std::string::String>>, axum::extract::Query<HashMap<std::string::String, std::string::String>>, HeaderMap) -> impl futures_util::Future<Output = Result<impl IntoResponse, HttpError>> {EntryApi::read_record}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `axum::routing::get`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:403:1
    |
403 | top_level_handler_fn!(get, GET);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `get`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<HashMap<std::string::String, std::string::String>>, axum::extract::Query<HashMap<std::string::String, std::string::String>>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {EntryApi::query}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:160:17
    |
160 |             get(EntryApi::query),
    |             --- ^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, axum::extract::Path<HashMap<std::string::String, std::string::String>>, axum::extract::Query<HashMap<std::string::String, std::string::String>>, HeaderMap) -> impl futures_util::Future<Output = Result<[type error], HttpError>> {EntryApi::query}`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `axum::routing::get`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:403:1
    |
403 | top_level_handler_fn!(get, GET);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `get`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>) -> impl futures_util::Future<Output = impl IntoResponse> {UiApi::redirect_to_index}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:163:51
    |
163 | ...pi_base_path), get(UiApi::redirect_to_index))
    |                   --- ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>) -> impl futures_util::Future<Output = impl IntoResponse> {UiApi::redirect_to_index}`
    |                   |
    |                   required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `axum::routing::get`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:403:1
    |
403 | top_level_handler_fn!(get, GET);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `get`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, Request<Body>) -> impl futures_util::Future<Output = Result<impl IntoResponse, HttpError>> {UiApi::show_ui}: Handler<_, _, _>` is not satisfied
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/reductstore-1.4.0-alpha.1/src/main.rs:164:23
    |
164 |         .fallback(get(UiApi::show_ui))
    |                   --- ^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(axum::extract::State<Arc<std::sync::RwLock<HttpServerComponents>>>, Request<Body>) -> impl futures_util::Future<Output = Result<impl IntoResponse, HttpError>> {UiApi::show_ui}`
    |                   |
    |                   required by a bound introduced by this call
    |
    = help: the following other types implement trait `Handler<T, S, B>`:
              <Layered<L, H, T, S, B, B2> as Handler<T, S, B2>>
              <MethodRouter<S, B> as Handler<(), S, B>>
note: required by a bound in `axum::routing::get`
   --> /home/atimin/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.6.18/src/routing/method_routing.rs:403:1
    |
403 | top_level_handler_fn!(get, GET);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `get`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0277, E0412, E0432, E0433.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `reductstore` due to 53 previous errors
error: failed to compile `reductstore v1.4.0-alpha.1`, intermediate artifacts can be found at `/tmp/cargo-installdDj8xX`
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- Cargo package, [PR-278](https://github.com/reductstore/reductstore/pull/278)
 
 ## [1.4.0-alpha.1] - 2023-05-22
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,7 +12,7 @@ repository = "https://github.com/reductstore/reductstore"
 readme = "README.md"
 keywords = ["database", "time-series", "blob", "storage", "reductstore"]
 categories = ["database", "storage"]
-include = ["src/**/*", "Cargo.toml", "Cargo.lock", "README.md", "LICENSE"]
+include = ["src/**/*", "Cargo.toml", "Cargo.lock", "build.rs", "README.md", "LICENSE"]
 
 [lib]
 crate-type = ["staticlib"]

--- a/README.md
+++ b/README.md
@@ -1,6 +1,3 @@
-
-
-
 # ReductStore
 
 [![GitHub release (latest SemVer)](https://img.shields.io/github/v/release/reductstore/reductstore)](https://github.com/reductstore/reductstore/releases/latest)
@@ -43,6 +40,7 @@ docker run -p 8383:8383 -v ${PWD}/data:/data reduct/store:latest
 or you can use cargo:
 
 ```shell
+apt install libssl-dev protobuf-compiler pkg-config
 cargo install reductstore
 RS_DATA_PATH=./data reductstore
 ```

--- a/build.rs
+++ b/build.rs
@@ -35,7 +35,11 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
             panic!("Failed to download Web Console: {}", resp.reason());
         }
     }
-    fs::write(format!("{}/console.zip", env::var("OUT_DIR").unwrap()), writer).expect("Failed to write console.zip");
+    fs::write(
+        format!("{}/console.zip", env::var("OUT_DIR").unwrap()),
+        writer,
+    )
+    .expect("Failed to write console.zip");
 
     // get build time and commit
     let build_time = chrono::DateTime::<chrono::Utc>::from(SystemTime::now())

--- a/build.rs
+++ b/build.rs
@@ -35,7 +35,7 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
             panic!("Failed to download Web Console: {}", resp.reason());
         }
     }
-    fs::write("src/asset/console.zip", writer).expect("Failed to write console.zip");
+    fs::write(format!("{}/console.zip", env::var("OUT_DIR").unwrap()), writer).expect("Failed to write console.zip");
 
     // get build time and commit
     let build_time = chrono::DateTime::<chrono::Utc>::from(SystemTime::now())

--- a/docs/README.md
+++ b/docs/README.md
@@ -30,9 +30,16 @@ ReductStore will be available on port http://127.0.01:8383 and store data in the
 
 ### Start With Cargo
 
+**Build Requirements:**
+
+* Minimal Rust version: 1.66.0
+* OpenSSL 3 or later
+* Protobuf 3.17.3 or later
+
 You can also install the database with cargo:
 
 ```
+sudo apt install libssl-dev protobuf-compiler pkg-config
 cargo install reductstore
 RS_DATA_PATH=./data reductstore
 ```

--- a/src/http_frontend/ui_api.rs
+++ b/src/http_frontend/ui_api.rs
@@ -104,9 +104,7 @@ mod tests {
             storage: Storage::new(PathBuf::from(data_path.clone())),
             auth: TokenAuthorization::new(""),
             token_repo: TokenRepository::new(PathBuf::from(data_path), ""),
-            console: ZipAssetManager::new(
-                include_bytes!(concat!(env!("OUT_DIR"), "/console.zip"))
-            ),
+            console: ZipAssetManager::new(include_bytes!(concat!(env!("OUT_DIR"), "/console.zip"))),
             base_path: "/".to_string(),
         };
 

--- a/src/http_frontend/ui_api.rs
+++ b/src/http_frontend/ui_api.rs
@@ -104,7 +104,9 @@ mod tests {
             storage: Storage::new(PathBuf::from(data_path.clone())),
             auth: TokenAuthorization::new(""),
             token_repo: TokenRepository::new(PathBuf::from(data_path), ""),
-            console: ZipAssetManager::new(include_bytes!("../asset/console.zip")),
+            console: ZipAssetManager::new(
+                include_bytes!(concat!(env!("OUT_DIR"), "/console.zip"))
+            ),
             base_path: "/".to_string(),
         };
 

--- a/src/main.rs
+++ b/src/main.rs
@@ -74,7 +74,8 @@ async fn main() {
         storage: Storage::new(PathBuf::from(data_path.clone())),
         auth: TokenAuthorization::new(&api_token),
         token_repo: TokenRepository::new(PathBuf::from(data_path), &api_token),
-        console: ZipAssetManager::new(include_bytes!("asset/console.zip")),
+        console: ZipAssetManager::new(include_bytes!(
+            concat!(env!("OUT_DIR"), "/console.zip"))),
         base_path: api_base_path.clone(),
     };
 

--- a/src/main.rs
+++ b/src/main.rs
@@ -74,8 +74,7 @@ async fn main() {
         storage: Storage::new(PathBuf::from(data_path.clone())),
         auth: TokenAuthorization::new(&api_token),
         token_repo: TokenRepository::new(PathBuf::from(data_path), &api_token),
-        console: ZipAssetManager::new(include_bytes!(
-            concat!(env!("OUT_DIR"), "/console.zip"))),
+        console: ZipAssetManager::new(include_bytes!(concat!(env!("OUT_DIR"), "/console.zip"))),
         base_path: api_base_path.clone(),
     };
 

